### PR TITLE
Refactor ChartBackground

### PR DESCRIPTION
**Pull request by billyjanitsch**
Writing the utils as component methods allows the linter to track props.

Need to figure out whether there's a way to salvage the tests.

### CODE REVIEW

**Comment by billyjanitsch** [comment]
Rebased.

File: src/ChartBackground/index.js [hunk 1]
```diff
@@ -14,149 +14,159 @@ import basicRender from '../canvas/basicRender'
...
-  width: PropTypes.number,
-  x: PropTypes.string,
-  xName: PropTypes.string,
-  xShowLabel: PropTypes.bool,
-  y: PropTypes.string,
-  yName: PropTypes.string,
-  yShowLabel: PropTypes.bool,
+  render() {
+    const {height, plotRect, theme, width, xShowLabel, yShowLabel} = this.props
+    const yText = this.getLabelText('y')
```
Comment (nabeel-) [inline src/ChartBackground/index.js]: What's the reasoning behind moving this out here? Isn't it more efficient to only call these functions if we need to?

Comment (billyjanitsch) [reply src/ChartBackground/index.js]: You're right, thanks.

**Comment by billyjanitsch** [comment]
I'm okay with merging without the tests, because they're mostly sanity checks which IMO are effectively covered by the sandbox.

**Pull request merged**